interfaces/cython/cantera/test/test_convert.py

```diff
@@ -717,3 +718,450 @@ def test_ch4_ion(self):
+        self.checkThermo(ctmlGas, yamlGas, [300, 500, 1300, 2000])
+
+    def test_sri_falloff_reaction(self):
+        ctml2yaml.convert(Path(self.test_data_dir).joinpath("sri-falloff.xml"),
+                          Path(self.test_work_dir).joinpath("sri-falloff.yaml"))
+        ctmlGas, yamlGas = self.checkConversion("sri-falloff")
+        self.checkThermo(ctmlGas, yamlGas, [300, 500, 1300, 2000])
+        self.checkKinetics(ctmlGas, yamlGas, [900, 1800], [2e5, 20e5])
+
+    def test_vpss_and_hkft(self):
+        ctml2yaml.convert(Path(self.test_data_dir).joinpath("pdss_hkft.xml"),
+                          Path(self.test_work_dir).joinpath("pdss_hkft.yaml"))
+
+        for name in ["vpss_gas_pdss_hkft_phase", "vpss_soln_pdss_hkft_phase"]:
+            ctmlPhase = ct.ThermoPhase("pdss_hkft.xml", name=name)
+            yamlPhase = ct.ThermoPhase("pdss_hkft.yaml", name=name)
+            # Due to changes in how the species elements are specified, the
+            # composition of the species differs from XML to YAML (electrons are used
+            # to specify charge in YAML while the charge node is used in XML).
+            # Therefore, checkConversion won't work and we have to check a few things
+            # manually. There are also no reactions specified for these phases so don't
+            # need to do any checks for that.
+            self.assertEqual(ctmlPhase.element_names, yamlPhase.element_names)
+            self.assertEqual(ctmlPhase.species_names, yamlPhase.species_names)
+            self.checkThermo(ctmlPhase, yamlPhase, [300, 500])
+
+    def test_lattice_solid(self):
+        ctml2yaml.convert(Path(self.test_data_dir).joinpath("Li7Si3_ls.xml"),
+                          Path(self.test_work_dir).joinpath("Li7Si3_ls.yaml"))
+        # Use ThermoPhase to avoid constructing a default Transport object which
+        # throws an error for the LatticeSolidPhase
+        basename = "Li7Si3_ls"
+        name = "Li7Si3_and_Interstitials(S)"
+        ctmlPhase = ct.ThermoPhase(basename + ".xml", name=name)
+        yamlPhase = ct.ThermoPhase(basename + ".yaml", name=name)
+        self.assertEqual(ctmlPhase.element_names, yamlPhase.element_names)
+        self.assertEqual(ctmlPhase.species_names, yamlPhase.species_names)
+        self.checkThermo(ctmlPhase, yamlPhase, [300, 500])
+
+    def test_margules(self):
+        ctml2yaml.convert(Path(self.test_data_dir).joinpath("LiKCl_liquid.xml"),
+                          Path(self.test_work_dir).joinpath("LiKCl_liquid.yaml"))
+        ctmlPhase, yamlPhase = self.checkConversion("LiKCl_liquid")
+        self.checkThermo(ctmlPhase, yamlPhase, [300, 500])
+
+    def test_idealsolidsoln(self):
+        with self.assertWarnsRegex(UserWarning, "SolidKinetics type is not implemented"):
+            ctml2yaml.convert(Path(self.test_data_dir).joinpath("IdealSolidSolnPhaseExample.xml"),
+                            Path(self.test_work_dir).joinpath("IdealSolidSolnPhaseExample.yaml"))
```

Comment:
Indentation issue here and in several other calls to `ctml2yaml.convert`.

Reply:
Indentation issue meaning the second line needs 2 more spaces? I think I might prefer adding a line break after `.convert(` and aligning the two `Path(...)` that way.

Reply:
Either of those changes would be fine. The latter should get the lines under 80 characters, which is nice.

Reply:
I've been using a linewidth of 88 characters as a compromise between 80 (which I think is too short) and 100 (which is too long for most people). This is also advocated by [Black](https://black.readthedocs.io/en/stable/)

Reply:
I'm sure I can find advocates for any number of different maximum line lengths. I don't think we need to enforce it rigidly, but we currently recommend 80 in [CONTRIBUTING.md](https://github.com/Cantera/cantera/blob/master/CONTRIBUTING.md).

Reply:
Sure, I picked that package because that's what I used to autoformat the script as I was writing it and I happen to agree with most of the choices they make in terms of formatting. Frankly, I think we should try to agree on a set of standards and enforce them with a bot or as part of the CI. A package such as YAPF is much more configurable, whereas Black is deliberately not configurable (one of their mottos is "Any color you want, as long as it's Black", the old Henry Ford quote), so YAPF might fit our needs better (for Python code anyways). Obviously this wouldn't get run over the whole codebase at once, but it would provide a level of consistency that would make PR reviews much easier. The trick is going to be making sure that the output from the bot/CI run is useful for people to actually make corrections, but it's something that many projects implement (some sort of automated linter, anyways).

Reply:
For what it's worth, expanding the line length limits to 88 in CONTRIBUTING.md would make sense, especially considering longish C++11 declarations (on top of an indent default of 4). Same for warnings and error messages.

Reply:
@ischoegl Thanks for the feedback. I'm going to open an issue (eventually) for discussion of how/if we want to add linting etc., as a part of which we could discuss changing the CONTRIBUTING guide.